### PR TITLE
mon: osd feature checks with 0 up osds

### PR DESCRIPTION
**Pull request by badone**
get_up_osd_features() returns 0 if no osds are up which causes feature
checks to fail.

Fixes: http://tracker.ceph.com/issues/21471

### CODE REVIEW

**Comment by badone** [comment]
@liewegas Hope something like this was what you had in mind?

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -8563,6 +8601,8 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
   } else if (prefix == "osd require-osd-release") {
     string release;
     cmd_getval(g_ceph_context, cmdmap, "release", release);
+    string sure;
+    cmd_getval(g_ceph_context, cmdmap, "sure", sure);
```
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: need ot add this to MonCommands.h

**Comment by badone** [comment]
@liewegas , still testing this but it should be pretty close?

**Comment by badone** [comment]
Of course it broke the unit tests. Working through the failures...

**Comment by badone** [comment]
Jenkins retest this please. Failure looks unrelated as far as I can tell and that test passes locally.

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-09-24_17:52:02-rados-wip-sage-testing-2017-09-24-1125-distro-basic-smithi/

**Pull request merged**